src/dev/i18n/extract_pug_messages.test.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { extractPugMessages } from './extract_pug_messages';
+
+const pugMessageBuffer = Buffer.from(`
+#{i18n('message-id', { defaultMessage: 'Default message', context: 'Message context' })}
+`);
+
+describe('extractPugMessages', () => {
+  it('extracts messages from pug template', () => {
+    for (const messageObject of extractPugMessages(pugMessageBuffer)) {
```

Comment:
__nit:__ looks like you just need ` const [messageObject] = extractPugMessages(pugMessageBuffer);` instead of all this `for` machinery?